### ISSUE TITLE
Copyright and license

### ISSUE THREAD

**Issue by nzakas** [estree/estree#214]
The ESTree repo doesn’t currently contain any copyright or license information. It would be good to decide on these as a way to signal to everyone how to use ESTree.

For copyright, we can probably go with “Copyright ESTree contributors” to start. 

For license, it seems like a Creative Commons license would be most appropriate as ESTree is not code. 

I think the big question in both cases is whether or not the original draft of ESTree was based on internal Mozilla documents or not. Maybe @dherman knows?

**Comment by caridy** [comment]
@syg might know as well!

**Comment by syg** [comment]
I'm not really qualified to talk about IPR issues like this. Are you asking if the original ESTree draft was based off documentation as opposed to based off code? Or are you asking if the original ESTree draft was based off Mozilla materials as opposed to non-Mozilla materials?

If you're asking the second question, my recollection is ESTree was originally based on SpiderMonkey's Reflect.parse, which reflected the internal SpiderMonkey AST.

**Comment by nzakas** [comment]
> Or are you asking if the original ESTree draft was based off Mozilla materials as opposed to non-Mozilla materials?

This. We know the format was based off the SpiderMonkey AST. What is unclear is if the first commit of the documentation to this repo was Mozilla-copyrighted or not. We’d like to apply an explicit copyright to this repo and would like it be accurate.

**Comment by nzakas** [comment]
Given that the SpiderMonkey AST is posted on [MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/Parser_API), and MDN has [these licenses](https://developer.mozilla.org/en-US/docs/MDN/About#Copyrights_and_licenses), it seems like the best copyright would be:

Copyright Mozilla Contributors and ESTree Contributors.

And the best license would be [Creative Commons Sharealike](https://creativecommons.org/licenses/by-sa/2.5/)

**Comment by nzakas** [comment]
Yes, what I'm saying is that ESTree is documentation that is based upon the documentation on MDN. Code doesn't really enter into the equation here.

**Comment by RReverser** [comment]
I understand that it's what you're saying, but I'm saying that I'm not sure it works that way and would rather consult someone who knows. After all, we've seen a pretty large legal disputes over whether APIs are subject to copyright :)

**Comment by nzakas** [comment]
@RReverser okay, how do you suggest we proceed?

**Comment by RReverser** [comment]
Probably best to write an email to Dave Herman (and cc us & @syg), maybe Dave will see it and will be able to direct us to people who can answer this. For now I'd suggest keeping this open and preserving status quo.

**Comment by nzakas** [comment]
@RReverser can you take lead on that?

**Comment by RReverser** [comment]
I was hoping you would since you were interested in this change TBH 😛

**Comment by nzakas** [comment]
Ah, and I figured you would because you are the lone dissenter. 😄 

I've actually tried reaching out to David several times in various mediums and have never heard back. I can keep trying, but at this point I don't know that it's worth the effort.

To circle back, it seems pretty clear from the [initial commit](https://github.com/estree/estree/commit/ed68e12245d90971140c67e8c0e7c4e4bc503ffb) that the documentation lived first on MDN and was copied into this repo. So I'd like to move forward with my proposed copyright and license as the Mozilla copyright and license are pretty clear. If there's any problem, we can always adjust in the future.

**Comment by dherman** [comment]
Hello! Here's what I can tell you about the history of the `Reflect.parse()` API and ESTree:

- I personally wrote the MDN documentation and the implementation of `Reflect.parse()` in SpiderMonkey. I was employed by Mozilla at the time, but I believe the copyright of both works is determined by the copyright notices on MDN (as @nzakas pointed out above) and the source files, respectively.
- The `Reflect.parse()` documentation was **not** derived from the source code. I personally designed the API and implemented it. The documentation came from my design, not from my implementation.

I'm not sure how relevant any of that is, though: regardless of the outcome of Oracle v Google, the MDN docs are licensed as Creative Commons, so the ESTree community has the explicitly-granted right to copy, share, and modify the API. And, well, Mozilla is not Oracle. 😛 

Bottom line, I don't think you should worry so much, and I think @nzakas's proposal makes a lot of sense: you're including all the copyright holders from the original work, you're honoring the terms of the CC license, and you're continuing with a consistent license.

But if you really want to be confident, you could reach out to [Mozilla legal](https://www.mozilla.org/about/legal/).

**Comment by RReverser** [comment]
@dherman Thanks for responding!

I think

> The Reflect.parse() documentation was not derived from the source code. 

This is the part that I was primarily legally worried about. Since it wasn't derived from source code, I also think it should be fine to assume it's Creative Commons as per MDN license then.

**Comment by nzakas** [comment]
Cool! Thanks everyone, I’ll work on putting together a PR to address this.
